### ISSUE TITLE
False positives: detected as unsupported features when using `--checkFeatures`

### ISSUE THREAD

**Issue by ItMaga** [yowainwright/es-check#364]
I found multiple issues where standard syntax is incorrectly flagged as newer ES features when running checks with the --checkFeatures flag.

### RegExp constructor detected as RegExpEscape

Standard `RegExp` constructor usage is flagged as `RegExpEscape`, which prevents passing checks for ES12 (even though the code works in much older environments).

#### Code to reproduce:

```javascript
// test-regexp.js
export function a(v) {
    return v.replace(RegExp(v, 'g'));
}
```

Command:
```
bash
npx es-check es12 './test-regexp.js' --checkFeatures
```

### Error constructor detected as ErrorCause

Using the standard `Error` constructor with a single string argument is incorrectly flagged as ErrorCause (ES2022). The code does not use the `{ cause: ... }` options bag, so it should be valid in older ES versions.

Code to reproduce:

```javascript
// test-error.js
export function a() {
    throw new Error('Cannot create key for mixed item');
}
```

Command:

```bash
npx es-check es12 './test-error.js' --checkFeatures
```

### Valid `async/await` detected as `TopLevelAwait`

Using `await` inside an `async` function is incorrectly flagged as `TopLevelAwait`. even though the `await` keyword is legally scoped inside the function and not at the module top level.

**Code to reproduce:**

```javascript
export async function a() {
    await Promise.all();
}
```

### Standard `in` operator detected as `ErgonomicBrandChecks`

The standard `in` operator used to check for property existence is incorrectly flagged as `ErgonomicBrandChecks` (which refers to `#private in object` checks from ES2022).

**Code to reproduce:**

```javascript
export function a() {
    const b = {};

    if ('c' in b) {
        //
    }
}
```

### Env

es-check: 9.5.2
node: 22.16.0

**Comment by yowainwright** [comment]
@ItMaga has the latest version fixed this issue for you? Thanks!
